### DISCUSSION TITLE
[Feature Request] Improve `jotai-scope` support

### DISCUSSION THREAD

**Discussion by kurateh** [jotaijs/jotai-devtools#137] [Ideas]
![image](https://github.com/jotaijs/jotai-devtools/assets/63982863/8c1a2f25-af7e-4cd5-9f23-acd1356d9fd8)

Due to the current implementation of jotai-scope, which creates a copy of an atom regardless of its scope (as shown in [this code](https://github.com/jotaijs/jotai-scope/blob/90e276c415893b4909efaa067d6d068c59721d5a/src/ScopeProvider.tsx#L37)), duplicated atoms appear in the dev-tool panel when an atom is accessed within multiple scopes.

So I would like to suggest the following enhancements:

1. Prevent displaying duplicated atoms in the devtool panel when they are duplicated by a `ScopeProvider` that they are not scoped in.
2. Include a feature in the devtool to show which `ScopeProvider` each atom is associated with.

Thank you for considering these improvements.

**Comment by dai-shi** [comment]
(This sounds like a tough one. Needs coordination with `jotai-scope` side.)

**Comment by arjunvegda** [comment]
Yeah, this is going to be a tough one. Should we explore the idea of "extensions" to Jotai DevTools? Extensions would allow third-party libraries to customize some behavior and the UI of the DevTools. 🤔 

Rough idea

```diff
import { DevTools } from 'jotai-devtools';
import 'jotai-devtools/styles.css';
+ import { devtoolsExtension } from 'jotai-scope';

const App = () => {
  return (
    <>
-      <DevTools />
+      <DevTools extensions=[devtoolsExtension] />
      {/* your app */}
    </>
  );
};

**Reply by dai-shi** [reply]
Sounds like a solid idea in the long term. Though, I'm not sure if there is a simpler solution to this specific case. Let's see if @yf-yang has any idea.

**Reply by arjunvegda** [reply]
Food for thought, for #1, would marking copied atoms as `debugPrivate=true` do the trick for time being? 🤔
